src/main/java/com/exasol/adapter/jdbc/BaseTableMetadataReader.java

```diff
@@ -50,20 +50,29 @@ public BaseTableMetadataReader(final Connection connection, final ColumnMetadata
     protected void mapOrSkipTable(final ResultSet remoteTables, final List<TableMetadata> translatedTables,
             final Optional<List<String>> selectedTables) throws SQLException {
         final String tableName = readTableName(remoteTables);
-        if (isTableIncludedByMapping(tableName) && isTableSelected(tableName, selectedTables)) {
-            mapOrSkipSupportedTable(remoteTables, translatedTables, tableName);
+        if (isTableIncludedByMapping(tableName)) {
+            mapSelectedTables(remoteTables, translatedTables, selectedTables, tableName);
         } else {
             skipUnsupportedTable(tableName);
         }
     }
 
+    private void mapSelectedTables(ResultSet remoteTables, List<TableMetadata> translatedTables,
```

Comment:
It would be `mapSupportedTables(...)` to be precise.

Reply:
fixed